### PR TITLE
Pack should pack symbol nupkg

### LINKED ISSUE #2116: Pack should pack pdb (if present) in a symbol nupkg
We do some work to ensure the compiled Razor assembly is added to a package. We're missing the equivalent feature to include the pdb in the symbol nupkg.

### PR DESCRIPTION
**Pull request by pranavkm**
* Update path calculation for BuiltProjectOutputGroupOutput to include full path. This matches
  the behavior of Microsoft.Common.targets.
* Add Razor symbols to DebugSymbolsProjectOutputGroupOutput

Fixes #2116

### CODE REVIEW

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 1]
```diff
@@ -320,16 +330,22 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+    DependsOnTargets="_ResolveRazorTargetPath;ResolveRazorGenerateInputs"
     Condition="'$(ResolvedRazorCompileToolset)'=='RazorSdk' and '$(RazorCompileOnBuild)'=='true'">
 
-    <PropertyGroup>
-      <RazorOutputPath Condition="'$(RazorOutputPath)'==''">$([MSBuild]::EnsureTrailingSlash('$(OutDir)'))</RazorOutputPath>
-    </PropertyGroup>
-
     <ItemGroup Condition="'@(RazorGenerate)'!= ''">
-      <BuiltProjectOutputGroupOutput Include="@(RazorIntermediateAssembly)" FinalOutputPath="$(RazorOutputPath)$(RazorTargetName).dll" />
+      <BuiltProjectOutputGroupOutput Include="%(RazorIntermediateAssembly.FullPath)" FinalOutputPath="$(RazorTargetPath)" />
```
Comment (pranavkm) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: This matches what Microsoft.Common.targets does - the output path is a full path

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 2]
```diff
@@ -486,6 +498,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <Error Text="A reference to Microsoft.AspNetCore.Razor.Design is required for this target to succeed. See todo:fwlink for instructions on how to resolve this." />
   </Target>
 
+  <Target Name="_ResolveRazorTargetPath">
+    <PropertyGroup>
+      <RazorOutputPath Condition="'$(RazorOutputPath)'==''">$([MSBuild]::EnsureTrailingSlash('$(OutDir)'))</RazorOutputPath>
+      <RazorTargetDir>$([MSBuild]::Escape($([MSBuild]::EnsureTrailingSlash($([System.IO.Path]::GetFullPath('$([System.IO.Path]::Combine('$(MSBuildProjectDirectory)', '$(RazorOutputPath)'))'))))))</RazorTargetDir>
```
Comment (pranavkm) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: This is based off of Common.targets too.

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 3]
```diff
@@ -429,7 +441,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Copy the Razor debug information file (.pdb), if any -->
     <Copy
       SourceFiles="@(_RazorDebugSymbolsIntermediatePath)"
-      DestinationFolder="$(RazorOutputPath)"
+      DestinationFolder="$(RazorTargetDir)"
```
Comment (pranavkm) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: I could leave this unchanged

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs [hunk 4]
```diff
@@ -313,7 +313,7 @@ public static void NuspecDoesNotContain(MSBuildResult result, string nuspecPath,
             }
         }
 
-        public static void NupkgContains(MSBuildResult result, string nupkgPath, string filePath)
+        public static void NupkgContains(MSBuildResult result, string nupkgPath, params string[] filePaths)
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs]: Can you just keep this as-is and call it multiple times? We don't really do this with the other asserts.

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs [hunk 5]
```diff
@@ -23,7 +23,7 @@ public class PackIntegrationTest : MSBuildIntegrationTestBase
             Assert.NuspecContains(
                 result,
                 Path.Combine("obj", Configuration, "ClassLibrary.1.0.0.nuspec"),
-                $"<file src=\"{Path.Combine("bin", Configuration, "netcoreapp2.0", "ClassLibrary.Views.dll")}\" " +
+                $"<file src=\"{Path.Combine(Project.DirectoryPath, "bin", Configuration, "netcoreapp2.0", "ClassLibrary.Views.dll")}\" " +
                 $"target=\"{Path.Combine("lib", "netcoreapp2.0", "ClassLibrary.Views.dll")}\" />");
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs]: This should test that the symbols are **not** included.

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs [hunk 6]
```diff
@@ -51,7 +78,7 @@ public class PackIntegrationTest : MSBuildIntegrationTestBase
             Assert.NuspecContains(
                 result,
                 Path.Combine("obj", Configuration, "ClassLibrary.1.0.0.nuspec"),
-                $"<file src=\"{Path.Combine("bin", Configuration, "netcoreapp2.0", "ClassLibrary.Views.dll")}\" " +
+                $"<file src=\"{Path.Combine(Project.DirectoryPath, "bin", Configuration, "netcoreapp2.0", "ClassLibrary.Views.dll")}\" " +
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs]: Is this right for a .nuspec? Is it **really** supposed to have the absolute path? are the other paths absolute paths?

Comment (pranavkm) [reply test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs]:
```xml
> cat .\obj\Debug\ClassLibrary.1.0.0.nuspec
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
  <metadata>
    <id>ClassLibrary</id>
    <version>1.0.0</version>
    <authors>ClassLibrary</authors>
    <owners>ClassLibrary</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>Package Description</description>
    <dependencies>
      <group targetFramework=".NETCoreApp2.0">
        <dependency id="Microsoft.AspNetCore.Razor.Runtime" version="2.1.0-preview2-t000" exclude="Build,Analyzers" />
        <dependency id="Microsoft.AspNetCore.Razor.Test.MvcShim" version="2.1.0-preview2-t000" exclude="Build,Analyzers" />
      </group>
    </dependencies>
  </metadata>
  <files>
    <file src="D:\work\Razor\test\testapps\ClassLibrary\bin\Debug\netcoreapp2.0\ClassLibrary.Views.dll" target="lib\netcoreapp2.0\ClassLibrary.Views.dll" />
    <file src="D:\work\Razor\test\testapps\ClassLibrary\bin\Debug\netcoreapp2.0\ClassLibrary.dll" target="lib\netcoreapp2.0\ClassLibrary.dll" />
  </files>
</package>
```

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs (hunk 4 above)
Comment (pranavkm) [reply test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs]: We unzip it to a fixed path which was causing multiple invocations to fail. `ExtractToDirectory` doesn't have a overwrite flag. This seemed like the least obtrusive solution. I can come up with something different if you feel strongly about this

Comment (rynowak) [reply test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs]: OH ok fine 👍 Add a comment

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs (hunk 6 above)
Comment (pranavkm) [reply test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/PackIntegrationTest.cs]: Bump?

**Pull request closed without merging**